sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs

```diff
@@ -52,11 +65,19 @@ private static HttpClient CreateDefaultClient()
 
         ServicePointHelpers.SetLimits(handler);
 
-        return new HttpClient(handler)
+        HttpClient defaultClient = new HttpClient(handler)
         {
             // Timeouts are handled by the pipeline
             Timeout = Timeout.InfiniteTimeSpan,
         };
+
+        lock (_lock)
```

Comment:
These 2 separate lock paths don't look thread-safe to me. I think there should be a check inside the lock to validate that SharedDefaultClient is still null.

Reply:
I see, because we set _httpClient outside the lock?  Let me look at this again.

Reply:
Oh, I think I can actually simplify this quite a bit - I think I was making it more complicated than it needs to be.  Will update shortly.

Reply:
Addressed - let me know if you have further feedback!

Reply:
Validating Azure.Core integration here: https://github.com/Azure/azure-sdk-for-net/pull/41462